### PR TITLE
feat(clickhouse handler): support user password in query parameters.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes https://github.com/datafuselabs/databend/issues/8955

### CODE REVIEW

File: src/query/service/src/servers/http/middleware.rs [hunk 1]
```diff
@@ -107,6 +108,36 @@ fn get_credential(req: &Request, kind: HttpHandlerKind) -> Result<Credential> {
...
+    if let (Some(name), Some(password)) = (user, key) {
+        let c = Credential::Password {
+            name: String::from_utf8(name.as_bytes().to_vec()).unwrap(),
+            password: Some(password.as_bytes().to_vec()),
+            hostname: client_ip,
+        };
+        Ok(c)
+    } else {
+        let query_str = req.uri().query().unwrap_or_default()?;
+        let query_params = serde_urlencoded::from_str::<HashMap<String, String>>(query_str)?;
```
[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/servers/http/middleware.rs:125]: where is `form_urlencoded` come from?

[the remark this answers is not in the document]
Comment (youngsofun) [reply src/query/service/src/servers/http/middleware.rs]: serde_urlencoded is used by poem for qurey string, so not intruduce new deps.

[the remark this answers is not in the document]
Comment (TCeason) [reply src/query/service/src/servers/http/middleware.rs]: > where is `form_urlencoded` come from?

https://docs.rs/form_urlencoded/1.1.0/form_urlencoded/

File: src/query/service/src/servers/http/middleware.rs [hunk 2]
```diff
@@ -107,6 +107,37 @@ fn get_credential(req: &Request, kind: HttpHandlerKind) -> Result<Credential> {
...
+        let (user, key) = (query_params.get("user"), query_params.get("password"));
+        if let (Some(name), Some(password)) = (user, key) {
+            Ok(Credential::Password {
+                name: name.clone(),
+                password: Some(password.as_bytes().to_vec()),
+                hostname: client_ip,
+            })
+        } else {
+            Err(ErrorCode::AuthenticateFailure(
+                "No  header or query parameters for authorization detected",
```
Comment (TCeason) [inline src/query/service/src/servers/http/middleware.rs:135]: > No  header or query parameters for authorization detected

No header or query parameters for authorization detected

**Pull request merged**